packages/web-components/src/utils/element-internals.ts

```diff
@@ -33,3 +33,36 @@ export function toggleState(elementInternals: ElementInternals, state: string, f
 
   elementInternals.states.delete(state);
 }
+
+/**
+ * A decorator for toggling the element internals of an element
+ *
+ * @internal
+ */
+export function toggleAttrState(target: any, attr: string): any {
+  // Stash the original method
+  const method = target[`${attr}Changed`];
+
+  // Replace the attrChanged method with a new one
+  target[`${attr}Changed`] = function (prev: string | boolean | undefined, next: string | boolean | undefined): void {
```

Comment:
Does this assignment affect super methods in class method overrides?

```ts
class SomeElement extends FASTElement {
    @toggleAttrState
    @attr
    something: string;

    somethingChanged(prev: string | undefined, next: string | undefined) {
        this.doSomething();
    }
}

class AnotherElement extends SomeElement {
    somethingChanged(prev: string | undefined, next: string | undefined) {

        super.somethingChanged(prev, next); // <-- here

        this.doSomethingElse();
    }
}
```

Reply:
Good callout. Had the same thought late last night. I'll investigate.

Reply:
Took some time to test this today and it all appears to be working.

1. I created a fake `FakeMessageBar` component

```ts
export class FakeMessageBar extends FASTElement {
  constructor() {
    super();
  }

  @toggleAttrState
  @attr
  public something?: string;

  somethingChanged(prev: string | undefined, next: string | undefined) {
    this.doSomething();
  }

  doSomething() {
    console.log('something changed', this.something);
  }
}
```

2. I made `MessageBar` extend `FakeMessageBar`...

```ts
export class MessageBar extends FakeMessageBar {
  public elementInternals: ElementInternals = this.attachInternals();

  constructor() {
    super();
    this.elementInternals.role = 'status';
  }

  somethingChanged(prev: string | undefined, next: string | undefined) {
    super.somethingChanged(prev, next); // <-- here

    this.doSomethingElse();
  }

  doSomethingElse() {
    console.log('something else changed', this.something);
  }

 ...
}
```

3. And then I was able to see make console logs and reflect in the console.

![image](https://github.com/user-attachments/assets/0b742a7c-3deb-4dfa-b663-1b8d934720fa)

I can/should write tests for this but I think it works as expected.